### PR TITLE
Update to rand 0.5

### PR DESCRIPTION
**Pull request by pitdicker**
Update all examples using rand to rand 0.5.

This version is not yet released, but will be soon. I think it is best to wait with merging this PR until then.

### CODE REVIEW

**Comment by pitdicker** [comment]
cc @dhardy

File: src/incomplete/rand.md [hunk 1]
```diff
@@ -93,9 +93,10 @@ struct Color { // RGB Color struct
...
 }
 
-// Implementing Rand for type Color
-impl Rand for Color {
-    fn rand<R: Rng>(rng: &mut R) -> Self {
+// Implement a Uniform random distribution for Color
+impl Distribution<Color> for Uniform {
+    #[inline]
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Point {
         Color {r: rng.next_f64(), b: rng.next_f64(), g: rng.next_f64()}
```
Comment (dhardy) [inline src/incomplete/rand.md]: `next_f64` doesn't exist any more

File: src/concurrency.md [hunk 2]
```diff
@@ -97,7 +98,7 @@ fn main() {
     // [2]
     vec.par_iter_mut().for_each(|p| {
         // [3]
-        *p = rand::weak_rng().gen_ascii_chars().take(5).collect()
+        *p = (0..5).map(|_| thread_rng().sample(Alphanumeric)).collect();
```
Comment (dhardy) [inline src/concurrency.md]: This creates a new `Rc` reference for each char. Still, maybe we can optimise that away.

File: src/incomplete/rand.md (hunk 1 above)
Comment (pitdicker) [reply src/incomplete/rand.md]: Examples in `inclomplete` are not tested... Will fix.

File: Cargo.toml [hunk 3]
```diff
@@ -29,7 +29,7 @@ memmap = "0.6"
 num = "0.1"
 num_cpus = "1.8"
 petgraph = "0.4"
-rand = "0.4"
+rand = "0.5"
```
Comment (AndyGauge) [inline Cargo.toml]: To get this to run I had to do rand="0.5.0-pre.1"  Do not change this, we will wait until the release to merge the PR.

File: src/basics.md [hunk 4]
```diff
@@ -246,22 +247,25 @@ fn main() {
...
+Randomly generates a tuple `(i32, bool, f64)` and variable of user defined type
+`Point`. Implements `Point` for [`rand::distributions::Uniform`] in order to
+allow random generation.
 
 ```rust
 extern crate rand;
 
-use rand::{Rng, Rand};
+use rand::Rng;
+use rand::distributions::{Distribution, Uniform};
```
Comment (AndyGauge) [inline src/basics.md]: I had to change this to `use rand::distributions::{Distribution, Standard};`

File: src/basics.md [hunk 5]
```diff
@@ -246,22 +247,25 @@ fn main() {
...
 
 #[derive(Debug)]
 struct Point {
     x: i32,
     y: i32,
 }
 
-impl Rand for Point {
-    fn rand<R: Rng>(rng: &mut R) -> Point {
+impl Distribution<Point> for Uniform {
```
Comment (AndyGauge) [inline src/basics.md]: This line error-ed with expected 1 type argument.  I was able to get it to compile with changing it to `Standard`.

**Comment by budziq** [comment]
Hi @pitdicker  Move to rand 0.5 was already merged with https://github.com/rust-lang-nursery/rust-cookbook/pull/418 thanks for your work!

**Pull request closed without merging**